master/internal/core.go

```diff
@@ -1312,6 +1355,7 @@ func (m *Master) Run(ctx context.Context, gRPCLogInitDone chan struct{}) error {
 		filepath.Join(m.config.Root, "swagger/determined/api/v1/api.swagger.json"))
 
 	m.echo.GET("/info", api.Route(m.getInfo))
+	m.echo.GET("/health", m.healthCheckEndpoint)
```

Comment:
question: would this run into one of grpc gateway limitations?

Reply:
Is the question why isn't this just an grpc route?

The reason is as far as I know, grpc has no way to return a non 200 status while returning a structured response

So I thought the status being 503 with a json body is important for the healthcheck and automation around it.

In GRPC you could return a 503 with some message but it won't be structured it is just going to be a string message. Hacking a structured response into the string message doesn't feel great. 

You could also always return 200 even for status's that are down but I think some people would just find the status easier to use and not care about trying to parse the message.

Reply:
> The reason is as far as I know, grpc has no way to return a non 200 status while returning a structured response


make sense. thanks
(not sure why my comment didn't post earlier)